examples/structural_mechanics/crash/train.py

```diff
@@ -0,0 +1,271 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os, sys, time, logging
+
+sys.path.insert(0, os.path.dirname(__file__))
+
+import hydra
+from hydra.utils import to_absolute_path, instantiate
+from omegaconf import DictConfig
+
+import torch
+from torch.cuda.amp import GradScaler, autocast
```

Comment:
Nit: for AMP API you should use `torch.amp` as `torch.cuda.amp` is deprecated. It will also remove "deprecated" warning when running this code.